Patient Registry System/PatientRegistrySystem.Services/UserService.cs

```diff
@@ -1,71 +1,59 @@
 ﻿using AutoMapper;
-using Microsoft.AspNetCore.Mvc;
-using PatientRegistrySystem.DB.Contexts;
 using PatientRegistrySystem.DB.Entities;
 using PatientRegistrySystem.DB.Repos;
 using PatientRegistrySystem.Domain.Dto;
-using System;
+using System.Collections.Generic;
 using System.Linq;
+using System.Threading.Tasks;
 
 namespace PatientRegistrySystem.Services
 {
-    public class UserService : Controller, IUserService
+    public class UserService : IUserService
     {
-        public IMapper Mapper { get; }
-        public IGenericRepository<User> userRepository { get; }
+        private readonly IMapper Mapper;
+        private readonly IUserRepository userRepository;
 
-        public UserService(IMapper mapper, IGenericRepository<User> UserGenericRepository)
+        public UserService(IMapper mapper, IUserRepository userRepository)
         {
-            Mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
-            userRepository = UserGenericRepository ?? throw new ArgumentNullException(nameof(UserGenericRepository));
+            Mapper = mapper;
+            this.userRepository = userRepository;
```

Comment:
when you have `_` at the beginning of a private prop, then `this.` it's not really necessary